### PR TITLE
Add C implementation for `@stdlib/math/base/special/exp10`

### LINKED ISSUE #734: [RFC]: Add C implementation for `@stdlib/math/base/special/exp10`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/exp10`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/exp10).

```c
double stdlib_base_exp10( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #734.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalpoly.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a final review.

File: lib/node_modules/@stdlib/math/base/special/exp10/README.md [hunk 1]
```diff
@@ -36,13 +36,13 @@ Evaluates the base `10` [exponential function][exponential-function].
 
 ```javascript
 var v = exp10( 3.0 );
-// returns 1000.0
+// returns ~1000.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/README.md]: Why are these return values considered approximate? They should be exact.

File: lib/node_modules/@stdlib/math/base/special/exp10/README.md [hunk 2]
```diff
@@ -75,6 +75,96 @@ for ( i = 0; i < 100; i++ ) {
...
+#include "stdlib/math/base/special/exp10.h"
+```
+
+#### stdlib_base_exp10( x )
+
+Evaluates the base `10` [exponential function][exponential-function].
+
+```c
+double out = stdlib_base_exp10( 3.0 );
+// returns ~1000.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/README.md]: Same comment as above.

File: lib/node_modules/@stdlib/math/base/special/exp10/src/main.c [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+*
+* @example
+* double out = stdlib_base_exp10( 2.0 );
+* // returns 100.0
+*/
+double stdlib_base_exp10( const double x ) {
+    double xc = x;
+    double px;
+    double xx;
+    double n;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/src/main.c]:
```suggestion
    int32_t n;
```

File: lib/node_modules/@stdlib/math/base/special/exp10/src/main.c [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    if ( xc > STDLIB_CONSTANT_FLOAT64_MAX_BASE10_EXPONENT ) {
+        return STDLIB_CONSTANT_FLOAT64_PINF;
+    }
+    if ( xc < STDLIB_CONSTANT_FLOAT64_MIN_BASE10_EXPONENT ) {
+        return 0.0;
+    }
+
+    // Express 10^x = 10^g 2^n = 10^g 10^( n log10(2) ) = 10^( g + n log10(2) )
+    px = stdlib_base_floor( ( LOG210 * xc ) + 0.5 );
+    n = px;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/src/main.c]:
```suggestion
    n = (int32_t)px;
```

File: lib/node_modules/@stdlib/math/base/special/exp10/src/main.c [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+* ```
+*/
+
+#include "stdlib/math/base/special/exp10.h"
+#include "stdlib/math/base/special/floor.h"
+#include "stdlib/math/base/special/ldexp.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/constants/float64/max_base10_exponent.h"
+#include "stdlib/constants/float64/min_base10_exponent.h"
+#include "stdlib/constants/float64/pinf.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/src/main.c:39]:
```suggestion
#include "stdlib/constants/float64/pinf.h"
#include <stdint.h>
```

File: lib/node_modules/@stdlib/math/base/special/exp10/README.md (hunk 2 above)
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/README.md]:
```suggestion
// returns 1000.0
```

File: lib/node_modules/@stdlib/math/base/special/exp10/README.md [hunk 6]
```diff
@@ -75,6 +75,96 @@ for ( i = 0; i < 100; i++ ) {
...
+
+out = stdlib_base_exp10( -9.0 );
+// returns ~1.0e-9
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/README.md]:
```suggestion
// returns 1.0e-9
```

File: lib/node_modules/@stdlib/math/base/special/exp10/README.md (hunk 1 above)
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/README.md]:
```suggestion
// returns 1000.0
```

File: lib/node_modules/@stdlib/math/base/special/exp10/README.md [hunk 7]
```diff
@@ -36,13 +36,13 @@ Evaluates the base `10` [exponential function][exponential-function].
 
 ```javascript
 var v = exp10( 3.0 );
-// returns 1000.0
+// returns ~1000.0
 
 v = exp10( -9.0 );
-// returns 1.0e-9
+// returns ~1.0e-9
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/README.md]:
```suggestion
// returns 1.0e-9
```

File: lib/node_modules/@stdlib/math/base/special/exp10/README.md [hunk 8]
```diff
@@ -36,13 +36,13 @@ Evaluates the base `10` [exponential function][exponential-function].
...
 
 v = exp10( 0.0 );
-// returns 1.0
+// returns ~1.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/exp10/README.md]:
```suggestion
// returns 1.0
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**